### ISSUE TITLE
polaris-select considers wrong item selected on initialisation

### ISSUE THREAD

**Issue by andrewpye** [smile-io/ember-polaris#242]
Noticed that on first initialisation, `polaris-select` always seems to consider the first option selected irrespective of the `value` passed in; this doesn't seem to be an issue in the React implementation.

In the GIF below, the selected value is `stripe`, but the control shows the checkmark next to the first item (`shopify`), which prevents that first option being clickable until another option has been clicked:

![polaris-select-bug](https://user-images.githubusercontent.com/5737342/47569579-9f610180-d93c-11e8-9d37-80f1c7b8d48e.gif)

**Comment by tomnez** [comment]
@andrewpye I looked into this and can't really see any big difference in implementation other than us using `value={{selectedOption}}` instead of `value={value}` to set the `value` attr of the `select` tag. But changing to `value={{value}}` doesn't make a difference either.

I tried waiting until `selectedOption` was set to render the select field, but that didn't help (thought maybe it could be a weird render issue with setting value after render or something). The only thing that seemed to work for me was explicitly adding a `selected` attr to the `option` tags:

```hbs
<option
  selected={{eq selectedOption opt.label}}
  value={{opt.value}}
  disabled={{opt.disabled}}
  hidden={{opt.hidden}}
>
```

but that goes against the React implementation so I'm not sure if we want to do that. Thoughts?

**Comment by andrewpye** [comment]
@tomnez thanks for investigating! I think we should definitely update to use `value={{value}}` instead of `value={{selectedOption}}` (that just seems like a small bug that slipped through). I've tested something like your suggested fix of passing `selected` explicitly on the `option` element and it seems to solve the problem. I'd say let's roll with that - I'd rather differ slightly from the React code and have the same functionality/lack of bugs, than match their code word-for-word 😉 The only thing I'd suggest would be to pass `selected={{eq opt.value value}}` since it seems slightly more prudent to match on the underlying values rather than on labels. I'll let you put a PR up to claim the glory for this one! 💪🔥

**Comment by andrewpye** [comment]
Closed in #245
